src/domain/platform/admin/opportunity/pages/InnovationFlow/OpportunityInnovationFlowView.tsx

```diff
@@ -1,21 +1,56 @@
 import { Grid } from '@mui/material';
 import React, { FC, useMemo } from 'react';
-import { useUrlParams } from '../../../../../../hooks';
-import { useOpportunityProfileInfoQuery } from '../../../../../../hooks/generated/graphql';
+import { useApolloErrorHandler, useUrlParams } from '../../../../../../hooks';
+import {
+  refetchOpportunityLifecycleQuery,
+  useHubLifecycleTemplatesQuery,
+  useOpportunityProfileInfoQuery,
+  useUpdateOpportunityInnovationFlowMutation,
+} from '../../../../../../hooks/generated/graphql';
 import Loading from '../../../../../../common/components/core/Loading/Loading';
-import EditLifecycle from '../../../../../platform/admin/templates/InnovationTemplates/EditLifecycle';
+import UpdateInnovationFlow from '../../../templates/InnovationTemplates/UpdateInnovationFlow';
 import OpportunityLifecycleContainer from '../../../../../../containers/opportunity/OpportunityLifecycleContainer';
+import { LifecycleType } from '../../../../../../models/graphql-schema';
+import { SelectInnovationFlowFormValuesType } from '../../../templates/InnovationTemplates/SelectInnovationFlowDialog';
 
 const OpportunityInnovationFlowView: FC = () => {
   const { hubNameId = '', opportunityNameId = '' } = useUrlParams();
+  const handleError = useApolloErrorHandler();
+
+  const { data: hubLifecycleTemplates } = useHubLifecycleTemplatesQuery({
+    variables: { hubId: hubNameId },
+  });
+  const innovationFlowTemplates = hubLifecycleTemplates?.hub?.templates?.lifecycleTemplates;
+  const filteredInnovationFlowTemplates = innovationFlowTemplates?.filter(
+    template => template.type === LifecycleType.Opportunity
+  );
 
   const { data: opportunityProfile } = useOpportunityProfileInfoQuery({
     variables: { hubId: hubNameId, opportunityId: opportunityNameId },
     skip: false,
```

Comment:
use skip here with the variables from line 17 - they might be undefined or in your case - empty strings.